tests/30rooms/13anonymousaccess.pl

```diff
@@ -272,6 +290,27 @@
       })->main::expect_http_403;
    };
 
+test "Anonymous users can get individual state for world_readable rooms after leaving",
+   requires => [ local_user_and_room_fixtures(), anonymous_user_fixture() ],
+
+   do => sub {
+      my ( $user, $room_id, $anonymous_user ) = @_;
+
+      Future->needs_all(
+         matrix_set_room_history_visibility( $user, $room_id, "world_readable" ),
+         matrix_set_room_guest_access( $user, $room_id, "can_join" ),
+      )->then( sub {
+         matrix_join_room( $anonymous_user, $room_id );
+      })->then( sub {
+         matrix_leave_room( $anonymous_user, $room_id );
+      })->then( sub {
+         do_request_json_for( $anonymous_user,
+            method => "GET",
+            uri    => "/api/v1/rooms/$room_id/state/m.room.member/".$user->user_id,
```

Comment:
I'm consciously using a different user's user ID, in case there's some weird special casing in a server which says something like "you can always find out your own membership state".